### ISSUE TITLE
two ingress hosts for one kuberntes service not work

### ISSUE THREAD

**Issue by kong62** [containous/traefik#6646]
Do you want to request a feature or report a bug?
Bug

What did you do?
when I set two hosts: ka.sit.hupu.io and kibana-ingress-sit to one kubernetes service kibana, it not working. when use one of them, it work ok.
```
# cat kibana-ingress.yaml                 
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  annotations:
    kubernetes.io/ingress.class: traefik
  name: kibana-ingress
  namespace: sit
spec:
  rules:
  - host: ka.sit.hupu.io
    http:
      paths:
      - backend:
          serviceName: kibana
          servicePort: 5601
        path: /

# cat kibana-ingress-sit.yaml 
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  annotations:
    kubernetes.io/ingress.class: traefik
  name: kibana-ingress-sit
  namespace: sit
spec:
  rules:
  - host: ka-sit.hupu.io
    http:
      paths:
      - backend:
          serviceName: kibana
          servicePort: 5601
        path: /
```

Output of traefik version: (What version of Traefik are you using?)
Version:      2.1.6

**Comment by kong62** [comment]
In Traefik Ingress,  in the host field , all "." Will be escaped to "-", i think this is a bug.

```
# kubectl logs -f -n traefik traefik-gcmw9 --tail=10
172.31.13.206 - - [07/Apr/2020:12:35:23 +0000] "POST //api HTTP/1.1" 200 176 "-" "-" 56844 "ka-sit-hupu-io@kubernetes" "http://172.31.12.100:8080" 24ms
```

**Comment by ddtmachado** [comment]
This was fixed on #6325 , and released on v2.2
I'm not sure if it will be backported to 2.1.x. ping @ldez

**Comment by ldez** [comment]
It's only on 2.2, FYI we manage only one minor version of a major version.

https://github.com/containous/traefik#release-cycle
